### PR TITLE
modified xml-api add-on to be compatible with HM-RASPBERRYMATIC

### PR DESCRIPTION
**Pull request by jens-maus**
modified installation scripts to be compatible with RaspberryMatic (HM-RASPBERRYMATIC) and applied some further optimizations like a proper cleanup upon uninstall and other minor changes.

### CODE REVIEW

File: xmlapi/cgi.tcl [hunk 1]
```diff
@@ -1335,30 +1335,30 @@ proc cgi_input {{fakeinput {}} {fakecookie {}}} {
...
-        return 1
-    } else {
-        set buffer $_cgi_read_line_buffer
-        set crlf ""
-        set _cgi_read_line_buffer ""
-        return [expr ([string length $buffer] > 0)]
-    }
+}
+
+set _cgi_read_line_buffer ""
```
Comment (ultrah) [inline xmlapi/cgi.tcl]: I assume this change is just formatting?

Comment (jens-maus) [inline xmlapi/cgi.tcl]: Exactly. Just converting Windows line feeds (CRLF) to Unix/Linux line feeds (LF).

**Comment by ultrah** [comment]
On my CCU2 this does not work. The XML-API does not show up as an installed adddon in the CCU UI and /addons/xmlapi returns 404

**Comment by jens-maus** [comment]
thanks for the hint, I will then see if I can update to be more compatible with a CCU2 as well.

**Comment by jens-maus** [comment]
Please download the updated 1.11 archive (https://github.com/jens-maus/XML-API/raw/master/xmlapi_addon_1.11.tar.gz) and see if it now works on a CCU2. Hopefully the changes I applied to the install script makes it installable on a plain CCU2 again.

**Comment by ultrah** [comment]
Working fine on my CCU2 now. Would be nice if someone with a CCU1 tries it out before we merge this. Or we introduce a develop branch

**Comment by jens-maus** [comment]
Well, the differences in installation between CCU1 and CCU2 are so minor in the update_script that I doubt that there should be any problem when trying to install it on a CCU1. However, if you know someone with a CCU1 feel free to ask him to try to install it. However, I don't think we should open a separate development branch for this.

**Comment by ultrah** [comment]
Sorry, I think I was unclear. I meant a /develop branch for anything that is not fully tested.

**Comment by ultrah** [comment]
By the way, I really think it is great that someone is contributing to this project again :)

**Comment by jens-maus** [comment]
Well, I probably won't change any deeper functionality of xml-api. The only thing I was interested in was getting this thing running on the RaspberryMatic I am running. 

Regarding the /develop branch I still think we should keep this pull request open until someone verified that it can be run on a CCU1 and if so you can merge it immediately. For the time being my fork of this repository can be meant to be the development branch.

**Comment by ultrah** [comment]
Sounds good. @Monty1979 any comment?

**Comment by Monty1979** [comment]
I'm testing on Thursday, the installation on the CCU1

**Comment by Monty1979** [comment]
Hello,

the addon 1.11 could be installed and working on the CCU1

**Comment by jens-maus** [comment]
Thanks for checking it on the CCU1. Then I hope it will be merged soon and perhaps someone will give me collaborator access to this repo so that I can directly commit changes in future here.

**Pull request merged**